desc/equilibrium/equilibrium.py

```diff
@@ -266,8 +283,28 @@ def __init__(
         self._M_grid = setdefault(M_grid, 2 * self.M)
         self._N_grid = setdefault(N_grid, 2 * self.N)
 
-        self._surface.change_resolution(self.L, self.M, self.N, sym=self.sym)
-        self._axis.change_resolution(self.N, sym=self.sym)
+        self._Nz = int(setdefault(Nz, self.surface._Nz))
+        self._Mz = int(setdefault(Mz, self.surface._Mz))
+        self._Lz = int(
+            setdefault(
+                Lz,
+                self.Mz if (self.spectral_indexing == "ansi") else 2 * self.Mz,
+            )
+        )
+
+        # RG: NFP, sym, Mz, Nz passed as kwargs, order is sacrosanct
+        self._surface.change_resolution(
+            self.L,
+            self.M,
+            self.N,
+            NFP=self._surface.NFP if hasattr(self._surface, "NFP") else self.NFP,
+            sym=self.sym,
+            Mz=self.Mz,
+            Nz=self.Nz,
+        )
+
+        # RG: If you must specify NFP here, specify the NFP of the axis
```

Comment:
what does this mean?

Reply:
Well, there is test which checks if axis.NFP is different from surface.NFP and eq.NFP so we must not specify eq.NFP here otherwise that test will fail.